api/src/main/java/org/pmiops/workbench/exfiltration/EgressRemediationService.java

```diff
@@ -323,54 +329,84 @@ private void logEventToJira(DbEgressEvent event, EgressRemediationAction action)
+            JiraContent.text("User Admin Console (Workbench Admin User): "),
+            user.map(
+                    u ->
+                        JiraContent.link(
+                            String.format(
+                                config.server.uiBaseUrl
+                                    + "/admin/users/"
+                                    + u.getUsername()
+                                        .replaceFirst(
+                                            "@" + config.googleDirectoryService.gSuiteDomain, ""))))
+                .orElse(JiraContent.text("unknown")),
+            JiraContent.text("\n\n")),
+        jiraEventDescriptionShort(event, action));
   }
 
-  private String jiraEventComment(DbEgressEvent event, EgressRemediationAction action) {
-    return "Additional egress detected\n\n" + jiraEventDescriptionShort(event, action);
+  private Stream<AtlassianContent> jiraEventComment(
+      DbEgressEvent event, EgressRemediationAction action) {
+    return Stream.concat(
+        Stream.of(JiraContent.text("Additional egress detected\n\n")),
+        jiraEventDescriptionShort(event, action));
   }
 
-  private String jiraEventDescriptionShort(DbEgressEvent event, EgressRemediationAction action) {
+  private Stream<AtlassianContent> jiraEventDescriptionShort(
+      DbEgressEvent event, EgressRemediationAction action) {
     Optional<DbWorkspace> workspace = Optional.ofNullable(event.getWorkspace());
     SumologicEgressEvent originalEvent =
         new Gson().fromJson(event.getSumologicEvent(), SumologicEgressEvent.class);
-    return String.format(
-        String.format("Action taken: %s\n\n", action)
-            + String.format(
+    return Stream.of(
+        JiraContent.text(String.format("Action taken: %s\n\n", action)),
+        JiraContent.text(
+            String.format(
                 "Terra Billing Project/workspace Namespace: %s\n",
-                workspace.map(DbWorkspace::getWorkspaceNamespace).orElse("unknown"))
-            + String.format("Google Project Id: %s\n\n", originalEvent.getProjectName())
-            + String.format(
-                "Detected @ %s",
-                Instant.ofEpochMilli(originalEvent.getTimeWindowStart()).atZone(jiraTimeZone))
-            + String.format(
+                workspace.map(DbWorkspace::getWorkspaceNamespace).orElse("unknown"))),
+        JiraContent.text(
+            String.format("Google Project Id: %s\n\n", originalEvent.getProjectName())),
+        JiraContent.text(
+            String.format(
+                "Detected @ %s\n",
+                Instant.ofEpochMilli(originalEvent.getTimeWindowStart()).atZone(jiraTimeZone))),
+        JiraContent.text(
+            String.format(
                 "Total egress detected: %.2f MiB in %d secs\n",
-                originalEvent.getEgressMib(), originalEvent.getTimeWindowDuration())
-            + String.format(
+                originalEvent.getEgressMib(), originalEvent.getTimeWindowDuration())),
```

Comment:
Just curious: did you actually trigger an egress to test this? Because I saw the time is 2018-7-31, and MiB are `nu`.

Reply:
No, I generated a fake event and POSTed it locally; I omitted some of the values which is why there are nulls. Sending a real egress event to my local machine is not feasible.

In hindsight, would have been smarter to copy and paste an event JSON from another env 🤷